### PR TITLE
feat: Custom assets

### PR DESCRIPTION
**Pull request by dgca**
Adds custom assets to block explorer including:

- [All assets page](https://block-explorer-git-feat-custom-assets-ironfish.vercel.app/assets)
- [Asset description page](https://block-explorer-git-feat-custom-assets-ironfish.vercel.app/assets/7c5e086c701b351516a73447f18407be7455779715c4e9d9ef3d851272f18f58)

### CODE REVIEW

File: pages/assets/[id].tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+              thickness="0.25rem"
+              speed="0.75s"
+            />
+          </HStack>
+        )}
+        <Box mb="8rem" />
+        {/* <Box mt="0.5rem" mb="2rem">
+          <h3>Asset History Chart</h3>
+        </Box>
+        <AssetTransactionChart /> */}
```
Comment (dgca) [inline pages/assets/[id].tsx:79]: Won't be adding this until the data is available from API.

File: package.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
   },
   "dependencies": {
     "@react-three/fiber": "^8.0.19",
+    "@tanstack/react-query": "^4.24.9",
```
Comment (dgca) [inline package.json:13]: Added react-query to make infinite queries easier to use. Would like to migrate fully to react-query later on.

File: components/CustomAssets/AssetHistory/AssetHistory.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        value={item.transaction_hash}
+        label={<HashView hash={item.transaction_hash} />}
+      />
+    ),
+  },
+  {
+    key: 'timestamp',
+    label: 'Timestamp',
+    render: item =>
+      formatInTimeZone(item.block_timestamp, 'UTC', 'yyyy/MM/dd hh:mm:ss aa'),
```
Comment (azhurauski-sc) [inline components/CustomAssets/AssetHistory/AssetHistory.tsx]: We have timestamp utils that is used to format all dates in explorer https://github.com/iron-fish/block-explorer/blob/update-ui-kit/utils/format/timestamp.ts#L10

Also there is TimeStamp component that can be used for compact view of date and time 
image

File: components/CustomAssets/CustomAssetsTable/CustomAssetsTable.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+          label={<HashView hash={identifier} />}
+        />
+      )
+    },
+  },
+  {
+    key: 'created_at',
+    label: 'Created',
+    render: item => {
+      return formatInTimeZone(
```
Comment (azhurauski-sc) [inline components/CustomAssets/CustomAssetsTable/CustomAssetsTable.tsx]: formatBlockTimestamp

File: pages/assets/[id].tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  const descriptions = useAssetDescriptions({
+    asset: id as string,
+  })
+
+  if (asset.error) {
+    throw asset.error
+  }
+
+  const assetDetails = useMemo(() => {
+    if (!asset.loaded) {
```
Comment (azhurauski-sc) [inline pages/assets/[id].tsx:27]: Can we use same loading state as in block view cards data for asset cards?
https://block-explorer-git-feat-custom-assets-ironfish.vercel.app/blocks/59391

File: pages/transaction/[hash].tsx [hunk 6]
```diff
@@ -229,37 +256,76 @@ const TransactionInfo = ({ data, loaded }) => {
...
-            label={card.label}
-            value={card.value(data)}
-            icon={card.icon}
-            isLoading={!loaded}
-          />
-        ))}
+        {TRANSACTION_INFO_CARDS.map(card => {
+          const value = card.value(data)
+          if (value === null) return null
+          return (
```
Comment (azhurauski-sc) [inline pages/transaction/[hash].tsx]: I'm not a big fun of inline if, but can we do 
```suggestion
          return value && (
```

File: components/CustomAssets/AssetHistory/AssetHistory.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  },
+]
+
+type Props = {
+  assetHistory: Array<AssetDescriptionType>
+}
+
+export function AssetHistory({ assetHistory }: Props) {
```
Comment (azhurauski-sc) [inline components/CustomAssets/AssetHistory/AssetHistory.tsx:49]: Just a question. Why simple functions?

Comment (dgca) [reply components/CustomAssets/AssetHistory/AssetHistory.tsx:49]: Mostly personal preference, but I also like that functions are hoisted, so you can put helper components at the bottom of a file.

I've seen that `React.FC` is [discouraged](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components/) by the community, so using `const Foo: React.FC<Props>` isn't really an issue anymore.

File: pages/assets/[id].tsx [hunk 8]
```diff
@@ -56,20 +56,9 @@ export default function AssetInfo() {
         <Box my="0.5rem">
           <h3>Asset History</h3>
         </Box>
-        {descriptions.loaded ? (
-          <AssetHistory assetHistory={descriptions.data} />
-        ) : (
-          <HStack justifyContent="center" py="6rem">
```
Comment (azhurauski-sc) [inline pages/assets/[id].tsx]: I didn't mean to remove that spinner thing it looks nice, I mean when you load block the cards has gray skeleton.
image

**Review by azhurauski-sc — APPROVED** [review]
Please check that build is passed before merge.

File: pages/assets/[id].tsx (hunk 8 above)
Comment (dgca) [reply pages/assets/[id].tsx]: Ah! Kk, I'll bring the spinner back and add the skeleton for the loading text.

**Pull request merged**